### PR TITLE
[Speedreader] Update to kuchikiki 0.8.2

### PR DESCRIPTION
**Pull request by rillian**
Update to our renamed fork to address unmaintained crate warnings about the `kuchiki` crate. Also cleans up some obsolete code.

*NB* This should probably wait until the rust build rework lands and be redone with the correct vendoring updates, but I wanted to get the source changes tested to make sure the new version will work for us.

Resolves https://github.com/brave/brave-browser/issues/29079

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by boocmp — APPROVED** [review]

**Comment by rillian** [comment]
It was waiting for the rust built rewrite to land. Now it just needs redoing on top of those changes, and reviewing and updating the tests. I won't be work on it until August, so anyone feel free to take it over.

**Comment by rillian** [comment]
I redid the patch on top of the new rust build integration. Haven't updated `html5ever` to match the version in kuchikiki's Cargo.toml, but it seems to work without that in local testing.

File: test/data/speedreader/rewriter/pages/news_pages/www.csmonitor.com/distilled.html [hunk 1]
```diff
@@ -62,7 +62,7 @@ <h2 class="title text-center">Why We Wrote This</h2>
...
...ongressional election in which voters are weighing candidates and a variety of issues. They say the abortion ruling may help Democrats cut in on Republicans’ margins somewhat, but not enough to pull off outright wins.</p><p>“People can latch onto anything for hope,” says Matt Gorman, former communications director for the National Republican Congressional Committee and current vice president at Targeted Victory. “The midterms are going to be dominated by the economy and inflation. That’s what people are feeling right now, and in every poll that’s what they’re caring about.”&nbsp;</p><p>To Mr. Gorman’s point, the <a href="https://www.monmouth.edu/polling-institute/reports/monmouthpoll_us_080322/" target="_self">recent Monmouth University poll</a> found that 24% of voters ranked economic policy as the most important issue in a congressional vote choice, followed by gun control policy (which has almost doubled in importance over the past three months) and abortion policy (which has actually fallen by 8 percentage points) tied at 17%. In third place is health control policy, followed by climate change and then immigration.&nbsp;</p><div class="in_story_embed embed-object embed-image ">
```
[the remark this answers is not in the document]
Comment (rillian) [reply test/data/speedreader/rewriter/pages/news_pages/www.csmonitor.com/distilled.html:65]: reviewdog complaints are all on formatting changes (attribute ordering) of test data, which is deliberately awful wild html.

[the remark this answers is not in the document]
Comment (thypon) [reply test/data/speedreader/rewriter/pages/news_pages/www.csmonitor.com/distilled.html:65]: all test data, gtg

**Comment by rillian** [comment]
@antonok-edm @bridiver Finally got this to pass CI. Ready for review again.

**Comment by rillian** [comment]
Replaced DEPS checkout with a vendored copy of the kuchikiki crate source; rebased past adjacent changes.

**Review by bridiver — APPROVED** [review]

**Pull request merged**